packages/content-conformance/src/engine.ts

```diff
@@ -39,6 +39,33 @@ export class ContentConformanceEngine {
     this.rules = opts.rules ?? []
   }
 
+  async loadDataFiles() {
+    if (!this.opts.dataFileGlobPattern) return
+
+    for await (const filepath of globbyStream(this.opts.dataFileGlobPattern, {
+      onlyFiles: true,
+      cwd: this.opts.root,
+    })) {
+      // If an array of filepaths are provided, only load the file if it matches one of the provided paths
+      if (
+        this.opts.files?.length &&
+        !this.opts.files.includes(String(filepath))
+      ) {
+        continue
+      }
+
+      const fullPath = path.join(this.opts.root, String(filepath))
+      const contents = await fs.promises.readFile(fullPath, 'utf-8')
+
+      const file = new DataFile({
+        cwd: this.opts.root,
+        path: String(filepath),
+        value: contents,
+      })
+      this.dataFiles.push(file)
+    }
+  }
```

Comment:
`loadDataFiles` and `loadContentFiles` are almost identical. I think we can likely consolidate the two methods going forward 💭